File: common/src/main/java/com/thewizrd/common/databinding/TextViewBindingAdapter.java
package com.thewizrd.common.databinding;

import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.databinding.BindingAdapter;

import com.thewizrd.common.utils.TextViewUtilsKt;

public class TextViewBindingAdapter {
    @BindingAdapter("clickable")
    public static void checkClickable(@NonNull TextView view, CharSequence text) {
        view.setClickable(TextViewUtilsKt.isTextTruncated(view));
    }
}

File: common/src/main/java/com/thewizrd/common/databinding/TextViewBindingAdapter.kt
package com.thewizrd.common.databinding

import android.widget.TextView
import androidx.core.view.doOnPreDraw
import androidx.databinding.BindingAdapter
import com.thewizrd.common.utils.isTextTruncated

object TextViewBindingAdapter {
    @JvmStatic
    @BindingAdapter("clickable")
    fun checkClickable(view: TextView, text: CharSequence?) {
        view.doOnPreDraw {
            it.isClickable = view.isTextTruncated()
        }
    }
}